app/controllers/my_module_repositories_controller.rb

```diff
@@ -159,20 +159,12 @@ def consume_modal
 
   def update_consumption
     module_repository_row = @my_module.my_module_repository_rows.find_by(id: params[:module_row_id])
-    module_repository_row.with_lock do
+
+    ActiveRecord::Base.transaction do
```

Comment:
with_lock creates transaction

Reply:
Added nested transaction support to the consume_stock method (via require_new: true). One transaction is for locked stock update, the one in controller is to make sure activity only gets created when it's successful.